### PR TITLE
Adding redirects for restructured Terraform Plugin Framework docs

### PR DESCRIPTION
**Pull request by bendbennett**
Closes: #2448 

### What

Added redirects for Terraform Plugin Framework docs

- `/plugin/framework/schemas` => `/plugin/framework/handling-data/schemas`
- `/plugin/framework/types` => `/plugin/framework/handling-data/attributes`
- `/plugin/framework/paths` => `/plugin/framework/handling-data/paths`
- `/plugin/framework/path-expressions` => `/plugin/framework/handling-data/path-expressions`
- `/plugin/framework/accessing-values` => `/plugin/framework/handling-data/accessing-values`
- `/plugin/framework/accessing-values#conversion-rules` => `/plugin/framework/handling-data/conversion-rules#converting-from-framework-types-to-go-types`
- `/plugin/framework/writing-state` => `/plugin/framework/handling-data/writing-state`
- `/plugin/framework/writing-state#conversion-rules` => `/plugin/framework/handling-data/conversion-rules#converting-from-go-types-to-framework-types`

### Why

Changes were made to rationalise the structure of the Terraform Plugin Framework docs and to make them simpler for users.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages.
- [x] **N/A** Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] **N/A** Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] **N/A** Pages with related content are updated and link to this content when appropriate.
- [x] **N/A** New pages have metadata (page name and description) at the top.
- [x] **N/A** New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] **N/A** New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] **N/A** UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Comment by bendbennett** [comment]
@BRKalow 👋 can you offer some guidance?

Is there a way to test these redirects before the PR is merged?

If I run the website locally and visit `localhost:3000/plugin/framework/schemas` I see a 404 but there's no redirect to `localhost:3000/plugin/framework/handling-data/schemas`.

Similarly, if I visit https://terraform-website-git-bendbennett-issues-2448-hashicorp.vercel.app/plugin/framework/schemas I see a 404 but it does not redirect to https://terraform-website-git-bendbennett-issues-2448-hashicorp.vercel.app/plugin/framework/handling-data/schemas

File: redirects.js [hunk 1]
```diff
@@ -25,4 +25,46 @@ module.exports = [
     destination: '/terraform/enterprise/sentinel/:path*',
     permanent: false,
   },
+  // Redirects for restructured Terraform Plugin Framework docs for GA release of the Framework
+  // - https://github.com/hashicorp/terraform-plugin-framework/pull/554
+  {
+    source: '/terraform/plugin/framework/schemas',
+    destination: '/terraform/plugin/framework/handling-data/schemas',
+    permanent: false,
```
Comment (BRKalow) [inline redirects.js:33]: Should the redirects here be permanent, or will the content move back to the old paths at some point?

File: redirects.js [hunk 2]
```diff
@@ -25,4 +25,46 @@ module.exports = [
...
+  {
+    source: '/terraform/plugin/framework/accessing-values',
+    destination: '/terraform/plugin/framework/handling-data/accessing-values',
+    permanent: false,
+  },
+  {
+    source: '/terraform/plugin/framework/accessing-values#conversion-rules',
+    destination: '/terraform/plugin/framework/handling-data/conversion-rules#converting-from-framework-types-to-go-types',
+    permanent: false,
+  },
```
Comment (BRKalow) [inline redirects.js:59]: Using anchor links is not valid in the source of the redirect, so this one won't work.

File: redirects.js [hunk 3]
```diff
@@ -25,4 +25,46 @@ module.exports = [
...
+  {
+    source: '/terraform/plugin/framework/writing-state',
+    destination: '/terraform/plugin/framework/handling-data/writing-state',
+    permanent: false,
+  },
+  {
+    source: '/plugin/framework/writing-state#conversion-rules',
+    destination: '/terraform/plugin/framework/handling-data/conversion-rules#converting-from-go-types-to-framework-types',
+    permanent: false,
```
Comment (BRKalow) [inline redirects.js:68]: Ditto here: using anchor links is not valid in the source of the redirect, so this one won't work.

File: redirects.js (hunk 1 above)
Comment (bendbennett) [reply redirects.js:33]: I don't think the content will revert back to the old paths so I've marked all of the redirects as `permanent: true`.

File: redirects.js (hunk 2 above)
Comment (bendbennett) [reply redirects.js:59]: I've removed this redirect as the rest of the `accessing-values` page needs to remain at `/terraform/plugin/framework/accessing-values`.

File: redirects.js (hunk 3 above)
Comment (bendbennett) [reply redirects.js:68]: I've removed this redirect as the rest of the `writing-state` page needs to remain at `/terraform/plugin/framework/writing-state`.

**Comment by BRKalow** [comment]
@bendbennett I don't think so, I can look into getting that fixed up. Regarding previewing the redirects, we don't currently support that yet in this repo. The redirects will get pulled into the Developer site (https://github.com/hashicorp/dev-portal) during the build process. The definitions look good to me. I think we're safe to merge when you're ready.

**Pull request merged**